source/draw/AbstractContext.ooc

```diff
@@ -0,0 +1,39 @@
+//
+// Copyright (c) 2011-2014 Simon Mika <simon@mika.se>
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+use ooc-base
+use ooc-draw
+use ooc-math
+
+AlignWidth: enum {
+	Nearest
+	Floor
+	Ceiling
+}
+
+AbstractContext: abstract class {
+	init: func
+	createMonochrome: abstract func (size: IntSize2D) -> Image
+	createBgr: abstract func (size: IntSize2D) -> Image
+	createBgra: abstract func (size: IntSize2D) -> Image
+	createUv: abstract func (size: IntSize2D) -> Image
+	createGpuImage: abstract func (rasterImage: RasterImage) -> Image
```

Comment:
`AbstractContext` will have another subclass called `CpuContext`, in which this method will return a `RasterImage`. I know this interface will require to be updated and I will do that, but for now I just wanted to keep the code in the *other project* unchanged.
Right now you can just merge this pull request and everything will work as usual.
This method is meant to return an image class *native* to given context implementation, so `GpuContext` will create `GpuImage`, while `CpuContext` will create `RasterImage`. So if you run with `CpuContext` the frame content will be a raster image, and that's what every client class will expect, because they will be replaced (via version blocks) by the cpu-only versions.
I guess we should simply call this method `createImage` or something. I ~~can~~ will do that in another pull request later on.